File: translation_output/no_test/gpt-4/codenet/python/java/p02909.java
Here's the equivalent Java code for the given Python script:

```java
import java.util.Scanner;

public class WeatherForecast {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String S = scanner.next();

        if (S.equals("Sunny")) {
            System.out.println("Cloudy");
        } else if (S.equals("Cloudy")) {
            System.out.println("Rainy");
        } else {
            System.out.println("Sunny");
        }

        scanner.close();
    }
}
```

This Java program does the same thing as the Python script. It reads a string from the user, checks the value, and prints out the next weather state accordingly. Note that the `equals()` method is used to compare strings for equality in Java, instead of the `==` operator which is used in Python.

File: translation_output/no_test/gpt-4/codenet/java/python/p02909.py
Sure, here's the equivalent Python code for the given Java code snippet:

```python
# No import is needed for input in Python

def main():
    str_input = input()

    if str_input == "Sunny":
        print("Cloudy")
    if str_input == "Cloudy":
        print("Rainy")
    if str_input == "Rainy":
        print("Sunny")

if __name__ == "__main__":
    main()
```

In Python, `input()` is used to get user input as a string. Also, print statements in Python do not require a call to `System.out.println`. The equality check between strings remains the same (`==` instead of `.equals()`). The overall structure of the Python code is also more simplified due to Python's syntax simplicity.
